packages/utils/telemetry-utils/src/errorLogging.ts

```diff
@@ -292,16 +300,16 @@ function getValidTelemetryProps(obj: any, keysToOmit: Set<string>): ITelemetryPr
             continue;
         }
         const val = obj[key];
+        const validProp = filterValidTelemetryProps(val);
         if (isTaggedTelemetryPropertyValue(val)) {
             props[key] = val;
```

Comment:
This behavior existed before, but since you are poking at it, I'd suggest fixing it.

Reply:
@vladsud -  Yeah we talked about that.  The thing is, it's supported in the type system to add a telemetry prop that's an array.  It's only the backdoor of slapping an array on a `LoggingError`.  So we landed on leaving it as-is.  If we formalize these flat arrays/objects for regular logging events (see AB#1104) then absolutely this needs to be updated as well.

Reply:
@chensixx - Maybe add a comment above the call to filterValidTelemetryProps explaining that we have to do this since any property could be on a `LoggingError` so we need to ensure we only log what is valid.

Reply:
@markfields , I think same logic applies to non-tagged properties as well.
I think it's less about arrays, it's more about REDACTED part - I can put random type and tag it, and it will go unchanged through the system (this includes arrays, but also objects with recursion that will likely explode down the road when someone tries to stringify it). It costs nothing to fix it and have uniform handling, whether we have it in type system or not.